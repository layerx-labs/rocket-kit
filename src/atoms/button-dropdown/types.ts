export interface ActionMenu<T> {
  id: string;
  value: string;
  visibilityFunc?: (data: T) => boolean;
  url?: string;
  action?: (
    e: React.MouseEvent<HTMLAnchorElement, MouseEvent> | React.ChangeEvent<HTMLSelectElement>,
    data: T | null
  ) => void;
}
