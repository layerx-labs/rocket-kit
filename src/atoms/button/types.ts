export type ButtonVariant = 'solid' | 'outline' | 'text';

export type ButtonColor =
  | 'primary'
  | 'danger'
  | 'info'
  | 'purple'
  | 'white'
  | 'black'
  | 'magic'
  | 'pulse';
