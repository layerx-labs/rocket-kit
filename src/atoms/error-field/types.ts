export type ErrorFieldColor = 'success' | 'red';
