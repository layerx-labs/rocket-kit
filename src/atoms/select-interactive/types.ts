import { MapHTMLAttributes, ReactNode } from 'react';
import NamedProps from 'react-select/dist/declarations/src/Select';

export interface TSelectInteractiveOption {
  value: string;
  label: string;
  [k: string]: any;
}

export interface TOptions extends TSelectInteractiveOption {
  icon?: string;
  customImage?: string;
}

export interface TCustomOptions {
  data: TOptions;
}

export interface OptionsGroup<T extends TSelectInteractiveOption> {
  label: string;
  options: ReadonlyArray<T>;
}

type TDivElement = Omit<
  MapHTMLAttributes<HTMLDivElement>,
  'onChange' | 'placeholder'
>;

type Options =
  | TSelectInteractiveOption
  | ReadonlyArray<TSelectInteractiveOption>;

export interface SelectInteractiveProps<T extends Options>
  extends TDivElement,
    Partial<Pick<NamedProps<T>, 'onInputChange'>> {
  name?: string;
  error?: string;
  clear?: boolean;
  multi?: boolean;
  search?: boolean;
  disabled?: boolean;
  placeholder?: ReactNode;
  formatGroupLabel?: boolean;
  value?: ReadonlyArray<TOptions> | TOptions | null | undefined;
  onChange: (values: Array<TOptions> | TOptions) => void;
  options: ReadonlyArray<TOptions | OptionsGroup<TOptions>>;
}
