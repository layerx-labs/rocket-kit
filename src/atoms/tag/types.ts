export type TagVariant = 'solid' | 'outline';
export type TagColor = 'primary' | 'orange' | 'danger' | 'info';
