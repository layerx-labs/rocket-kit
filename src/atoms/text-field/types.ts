export type TextFieldType =
  | 'text'
  | 'url'
  | 'password'
  | 'date'
  | 'time'
  | 'number'
  | 'email';
