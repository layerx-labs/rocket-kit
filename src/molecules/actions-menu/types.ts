export type ActionMenuType = 'danger' | 'default';

export interface ActionMenu<T> {
  id: string;
  type: ActionMenuType;
  value: string;
  visibilityFunc?: (data: T) => boolean;
  url?: string;
  action?: (
    e: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
    data: T | null
  ) => void;
}
