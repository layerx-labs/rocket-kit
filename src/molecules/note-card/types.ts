export type NoteColor = 'grey' | 'green' | 'red';
