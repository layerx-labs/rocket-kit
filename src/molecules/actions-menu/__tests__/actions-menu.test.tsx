import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import 'jest-styled-components';
import { ActionsMenu } from '../../..';
import { ActionMenu } from '../types';

describe('Actions Menu', () => {
  it('renders open menu', () => {
    const actions: ActionMenu<string>[] = [
      {
        id: 'actionDetails',
        type: 'default',
        value: 'Go to Details',
        action: () => {},
      },
    ];

    const { asFragment } = render(
      <ActionsMenu<string> actions={actions} startsOpen />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders closed menu', () => {
    const actions: ActionMenu<string>[] = [
      {
        id: 'actionDetails',
        type: 'default',
        value: 'Go to Details',
        action: () => {},
      },
    ];

    const { asFragment } = render(<ActionsMenu<string> actions={actions} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('has action in open menu', () => {
    const actionText = 'Go to Details';
    const actions: ActionMenu<string>[] = [
      {
        id: 'actionDetails',
        type: 'default',
        value: actionText,
        action: () => {},
      },
    ];

    render(<ActionsMenu<string> actions={actions} startsOpen />);
    screen.getByText(actionText);
  });

  it('has action in open menu with multiple items', () => {
    const actionText = 'Go to Details';
    const actions: ActionMenu<string>[] = [
      {
        id: 'actionDetails',
        type: 'default',
        value: actionText,
        action: () => {},
      },
      {
        id: 'action2',
        type: 'default',
        value: 'Edit',
        action: () => {},
      },
      {
        id: 'action3',
        type: 'danger',
        value: 'Delete',
        action: () => {},
      },
    ];

    render(<ActionsMenu<string> actions={actions} startsOpen />);
    screen.getByText(actionText);
  });

  it('can open menu with click', () => {
    const actionText = 'Go to Details';
    const actions: ActionMenu<string>[] = [
      {
        id: 'actionDetails',
        type: 'default',
        value: actionText,
        action: () => {},
      },
    ];

    render(<ActionsMenu<string> actions={actions} />);
    const foundButton = screen.getByTestId('action-menu-button');
    userEvent.click(foundButton);
    screen.getByText(actionText);
  });

  it('calls action callback on click', () => {
    interface ActionData {
      name: string;
      type: string;
    }

    const data: ActionData = {
      name: 'hello',
      type: 'test',
    };

    const onClick = jest.fn();
    const actionText = 'Go to Details';
    const actions: ActionMenu<ActionData>[] = [
      {
        id: 'actionDetails',
        type: 'default',
        value: actionText,
        action: onClick,
      },
    ];

    render(
      <ActionsMenu<ActionData> actions={actions} startsOpen data={data} />
    );
    const action = screen.getByText(actionText);
    userEvent.click(action);
    expect(onClick).toBeCalledTimes(1);
    expect(onClick).toBeCalledWith(expect.anything(), data);
  });
});
