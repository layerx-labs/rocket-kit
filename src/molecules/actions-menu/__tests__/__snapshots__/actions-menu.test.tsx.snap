// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions Menu renders closed menu 1`] = `
<DocumentFragment>
  <div
    class="sc-crrsfI cnURua"
  >
    <button
      aria-label=""
      class="sc-gsTCUz dQVqYO button"
      color="dark"
      data-testid="action-menu-button"
      value=""
    >
      <svg
        class="icon"
        viewBox="0 0 32 32"
      >
        <path
          d="M16 10.667c1.473 0 2.667-1.194 2.667-2.667s-1.194-2.667-2.667-2.667c-1.473 0-2.666 1.194-2.667 2.666v0c0.004 1.471 1.196 2.662 2.666 2.667h0zM16 13.333c-1.473 0-2.667 1.194-2.667 2.667s1.194 2.667 2.667 2.667c1.473 0 2.667-1.194 2.667-2.667v0c-0.005-1.471-1.196-2.662-2.666-2.667h-0zM16 21.333c-1.473 0-2.667 1.194-2.667 2.667s1.194 2.667 2.667 2.667c1.473 0 2.667-1.194 2.667-2.667v0c-0.005-1.471-1.196-2.662-2.666-2.667h-0z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Actions Menu renders open menu 1`] = `
<DocumentFragment>
  <div
    class="sc-crrsfI cnURua"
  >
    <button
      aria-label=""
      class="sc-gsTCUz dQVqYO button"
      color="dark"
      data-testid="action-menu-button"
      value=""
    >
      <svg
        class="icon"
        viewBox="0 0 32 32"
      >
        <path
          d="M16 10.667c1.473 0 2.667-1.194 2.667-2.667s-1.194-2.667-2.667-2.667c-1.473 0-2.666 1.194-2.667 2.666v0c0.004 1.471 1.196 2.662 2.666 2.667h0zM16 13.333c-1.473 0-2.667 1.194-2.667 2.667s1.194 2.667 2.667 2.667c1.473 0 2.667-1.194 2.667-2.667v0c-0.005-1.471-1.196-2.662-2.666-2.667h-0zM16 21.333c-1.473 0-2.667 1.194-2.667 2.667s1.194 2.667 2.667 2.667c1.473 0 2.667-1.194 2.667-2.667v0c-0.005-1.471-1.196-2.662-2.666-2.667h-0z"
        />
      </svg>
    </button>
    <ul
      data-testid="ul-action-menu"
    >
      <li
        data-testid="li-action-menu"
      >
        <a
          href=""
        >
          <span>
            Go to Details
          </span>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
